src/components/ec-metroline/ec-metroline.story.js

```diff
@@ -0,0 +1,165 @@
+import EcMetroline from './ec-metroline.vue';
+import EcMetrolineItem from '../ec-metroline-item';
+import EcIcon from '../ec-icon/ec-icon.vue';
+import EcBtn from '../ec-btn/ec-btn.vue';
+import * as MetrolineItemStatus from '../../enums/metroline-item-status';
+
+export default {
+  title: 'Metroline',
+  component: EcMetroline,
+};
+
+const Template = (args, { argTypes }) => ({
+  props: Object.keys(argTypes),
+  components: {
+    EcMetroline,
+    EcMetrolineItem,
+    EcIcon,
+    EcBtn,
+  },
+  template: `
+  <div class="tw-m-24 tw-flex tw-justify-center">
+    <ec-metroline class="tw-max-w-screen-md">
+        <ec-metroline-item
+          :id="1"
+          :badgeText="badgeText"
+        >
+          <template #heading>
+            <span class="tw-mr-16">
+              Transaction Information
+            </span>
+          </template>
+
+          <template #sub-heading>
+            <span class="tw-flex tw-items-center tw-mr-16">
+              <ec-icon name="simple-check" class="tw-fill-current tw-mr-8" :size="16" />
+              Amount fully allocated
+            </span>
+          </template>
+
+          <template #header-cta="{ goTo }">
+            <a href="#" class="tw-flex tw-items-center" @click.prevent.stop="goTo">
+              <ec-icon name="simple-edit" class="tw-fill-current tw-mr-8" :size="24" />
+              Edit
+            </a>
+          </template>
+
+          <template #header-cta-complete>
+            <a href="#" @click.prevent.stop class="tw-flex tw-items-center">
+              Add Description
+            </a>
+          </template>
+
+          <template #main="{ status }">
+            <div v-if="status === MetrolineItemStatus.COMPLETED" class="tw-bg-gray-7 tw-min-h-104 tw-p-24">
```

Comment:
I think a better solution would be bind isCompleted, isNext, isActive computed props from the component so you don't have to recalculate them here again. Especially MetrolineItemStatus enum doesn't have to be added to the data of the component's template.

Don't forget to bind them for every slot, just like goTo, goToNext, isReadOnly, etc.

Reply:
Removed status since we will bind the other 3